### ISSUE TITLE
new to suitecrm and your module

### ISSUE THREAD

**Issue by f1-outsourcing** [serfreeman1337/asterlink#9]
The readmes is are not really clear how this supposed to work if you are quite new. Currently I am using the odoo crap and that connects directly to asterisk for similar functionality (although as everything there, buggy). So I do not really get why it is necessary to have a process running, or is the zip file enough?

**Comment by serfreeman1337** [comment]
It's designed such way that process communicates with asterisk via ami, it monitors for calls/requests and then sends info to suitecrm. On the suitecrm side module is need for frontend tweaks (call card/click2call) and api for call logging. 

asterisk <-- (ami) --> asterlink process <-- (http/ws via module) --> suitecrm

So running process is necessarry.

- First [install asterlink](https://github.com/serfreeman1337/asterlink#install)
- Then [integrate asterlink with suticrm](https://github.com/serfreeman1337/asterlink/blob/master/README_suitecrm.md#basic-setup)

**Comment by f1-outsourcing** [comment]
What would be the minimum required permissions for the asterisk manager, currently I just grabbed something from the internet 

```
read = all,system,call,user,dtmf
write = all,system,call
```
because with these asterlink would not start.

```
read =
write = originate,reporting
```

**Comment by f1-outsourcing** [comment]
method=POST suite=true url="http://crm.xxxxxxx.xxx/instance/index.php?entryPoint=AsterLinkEntryPoint&action=create_call_record"
TRAC[0570] 404 Not Found                     


http://crm.xxxxxx..xxx/instance/index.php also gives error

**Comment by f1-outsourcing** [comment]
in the module configuration in suitecrm I would also make better clear what settings are related to suitecrm side and what settings are related to asterlink

Also put working default values. It is easier to replace hostnames than figure out what needs to be entered in empty fields, if it needs to be the hostname of asterlink or suitecrm or voip server.

**Comment by serfreeman1337** [comment]
> What would be the minimum required permissions for the asterisk manager, currently I just grabbed something from the internet
```
read = call,dialplan,originate
write = system,originate
```

> http://crm.xxxxxx..xxx/instance/index.php also gives error
```
suitecrm:
   url: http://my.suitecrm/instance/
...
```
It's an example for suitecrm installed in subfolder **/instance/** in the webserver fodler. 
Set it to **http://crm.xxxxxx..xxx/**, if suitecrm installed in the root of webserver folder (htdocs/html).

> enpoint settings? endpoint in suitecrm or end point of asterlink???

It's asterlink endpoint settings. User's browser will connect to asterlink (for call card or click2call) using those URL.
I.e. asterlink needs to be available for user, either directly or by apache2/nginx proxy.

**Comment by f1-outsourcing** [comment]
yes removing the /instance/, i have the call logged.

Is there also functionality that asterisks gets the Caller name from suitecrm so it can add this to the sip callerid, so it the name will be displayed on the phone?

**Comment by serfreeman1337** [comment]
No at this moment. I will add it in next release.

**Comment by f1-outsourcing** [comment]
> No at this moment. I will add it in next release.

COOL! :)

**Comment by f1-outsourcing** [comment]
> No at this moment. I will add it in next release.

Is the new release also for suitecrm 8?

**Comment by f1-outsourcing** [comment]
I am not getting the calls connected to accounts/contacts, I assume this can be realized with the relationship settings?  But I can't seem to find what would be a good module to choose. Eg. I have no idea what the difference is between 'Related to Leads' or 'Leads'.

**Comment by serfreeman1337** [comment]
<details>
  <summary>Module configuration example</summary>
  
   ![image](https://community.suitecrm.com/uploads/default/original/2X/8/89e1505e32f1aeb7a9f683c36696f16b7c479c42.png)
</details>

If i recall correctly, "related to" shows that it's a primary module in relation to "Calls" module.  
https://docs.suitecrm.com/admin/administration-panel/studio/#_relationships

It can be verified by looking at "Calls" module relationships in studio.
For contacts just "Contacts" needs to be selected, because "Calls" is primary in that case, but for leads it's otherwise, since "Leads" is primary in relationship.

Contacts: Calls -> Contacts (Calls is primary module)
Related to Leads: Calls <- Leads (Leads is primary module)
<sub><sup>Also I remember that there was a hardcoded relationships in suitecrm regarding to "Contacts" module, so "Related to Contacts" may also be displayed, but "Contacts" needs to be selected to match "Calls" primary module module relationships in studio.</sub></sup>

**Comment by f1-outsourcing** [comment]
suitecrm8 did not work properly, even when I installed the zip file. I noticed at least one form input variable not being verified and just dumped in a htaccess file.  Also this speed is still not fixed in 8.1.2 as this Paul is complaining about in 2021[1], it was even annoying to me. So I am back on suitecrm7.

[1]
https://www.youtube.com/watch?v=Knt6U5ga4nI

**Comment by serfreeman1337** [comment]
asterlink process logs correct time for date_start, date_end
ie:
```
time="2022-06-26T19:44:14+02:00" level=trace msg="map[data:map[asterlink_call_seconds_c:17 date_end:2022-06-26 17:44:14 duration_hours:0 duration_minutes:1 status:]
```
but in suitecrm it's 00 seconds ?

**Comment by f1-outsourcing** [comment]
Yes the above is from the suitecrm database. You can notice the date_modified, so the modification of the records is being received. 

```
date_entered	date_modified	modified_user_id	created_by	description	deleted	assigned_user_id	duration_hours	duration_minutes	date_start	date_end
2022-06-28 10:07:20	2022-06-28 10:08:12	(null)		(null)	false		0	0	2022-06-28 10:07:00	2022-06-28 10:07:00
```

This is logged by asterlink, look how the time is the same as the modified date field, so I guess the suitecrm module does not update it?

`e msg="map[data:map[asterlink_call_seconds_c:52 date_end:2022-06-28 10:08:12 `

**Comment by f1-outsourcing** [comment]
You also have this? Or I am just the lucky only one having this?

**Comment by serfreeman1337** [comment]
> Is there also functionality that asterisks gets the Caller name from suitecrm so it can add this to the sip callerid, so it the name will be displayed on the phone?

This can be achieved with "CID Superfecta" freepbx module. Configure "SugarCRM" Data Source in superfecta scheme and enable superfecta lookup for inbound route.
